src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -80,19 +80,27 @@ public IEnumerable<string> SupportedVersions
         /// <inheritdoc/>
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
-            var detectionResult = _detector.Detect(new DetectorContext
+            try
             {
-                SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
-            });
+                var detectionResult = _detector.Detect(new DetectorContext
+                {
+                    SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
+                });
+                if (detectionResult == null)
+                {
+                    return null;
+                }
 
-            if (detectionResult == null)
+                var version = ResolveVersion(detectionResult.PlatformVersion);
+                detectionResult.PlatformVersion = version;
+                return detectionResult;
+            }
+            catch (InvalidProjectFileException e)
             {
-                return null;
+                _logger.LogError(e, "InvalidProjectFileException caught: Either could not find the .NET Core project file" +
```

Comment:
`LogError` should have an overload which takes in an exception, so for the message part you can just say like 'error occurred while detecting for .net core application(s)'.